intake/catalog/base.py

```diff
@@ -438,15 +448,19 @@ def fetch_page(self, page_offset):
                 "Failed to fetch page of entries {}-{}."
                 "".format(page_offset, page_offset + self._page_size)), err)
         info = msgpack.unpackb(response.content, **unpack_kwargs)
-        page = {source['name']: RemoteCatalogEntry(
-            url=self.url,
-            getenv=self.getenv,
-            getshell=self.getshell,
-            auth=self.auth,
-            http_args=self.http_args,
-            page_size=self._page_size,
-            **source)
-            for source in info['sources']}
+        page = {}
+        for source in info['sources']:
+            user_parameters = source.get('user_parameters', [])
+            # TODO Do something with self._parameters.
```

Comment:
I'm confused about how to merge the `source['user_parameters']` (from the server) with `self._parameters` that would have come in through `open_remote`. Are they expected to just be the same thing?

Reply:
To elaborate: The code here replicates the pre-existing behavior. Previously, `parameters` were passed in by `open_remote` and then dropped on the floor by `Catalog`'s `**kwargs`. Now `parameters` is picked off by `RemoteCatalog` and...dropped on the floor. The question is how/whether to merge those parameters with those in `source`.

Reply:
I think it is OK to leave like this until we know we need it. As you say, there is more than one way to pass these things around, I expect some evolution.